h/services/annotation_moderation.py

```diff
@@ -23,9 +23,17 @@ def all_hidden(self, annotation_ids: str) -> set[str]:
         return {a.id for a in query if a.is_hidden}
 
     def set_status(
-        self, annotation: Annotation, user: User, status: ModerationStatus | None
+        self,
+        annotation: Annotation,
+        status: ModerationStatus | None,
+        user: User | None = None,
     ) -> None:
         """Set the moderation status for an annotation."""
+
+        if not user:
+            # If no user is provided assume this is an automatic transition triggered by the annotation author
+            user = self._user_service.fetch(annotation.userid)
```

Comment:
Alternatively we could make AnotationLog.moderation_id nullable and don't bother with this.

Then moderation_id == None would implicitly mean a change made by the annotation author.

Reply:
Hmm. I actually hadn't considered that we might want _automatic_ moderation state changes to show in the moderation log as well, but makes sense. For these automatic changes `ModerationLog.moderator_id = None` actually makes more sense to me than setting it to the annotation's author's ID. The author didn't change the moderation status of the annotation. All the author did was create or update the annotation. The _system_ automatically changed the moderation status.

Reply:
Yeah I agree. 

Made this other PR to make moderator_id nullable: https://github.com/hypothesis/h/pull/9534